### ISSUE TITLE
ECO mode cannot be activated.

### ISSUE THREAD

**Issue by MiJo112** [marq24/ha-goecharger-api2#7]
### Checklist

- [X] I have installed the **latest** release (or BETA) version of the integration and home assistant.
- [X] I have prepared DEBUG log output (for technical issues) | In most of the cases of a technical error/issue I would have the need to ask for DEBUG log output of the integration. There is a short [tutorial/guide 'How to provide DEBUG log' here](https://github.com/marq24/ha-senec-v3/blob/master/docs/HA_DEBUG.md)
- [X] I understand the SHIT-__IN__ - SHIT-__OUT__ principle (in case of implausible data) | It will happen, that the data that is displayed by this integration does not make much sense (to you) - aka the data is not plausible. __Of course__ it could be the case, that something in this integration has been messed up - but so far - in all reported issues the root cause of implausible data was/is, that the go-eCharger device itself already provided this data [you can check this by directly requesting the attribute from the wallbox]
- [X] I confirm it's really an issue | In the case that you want to understand the functionality of a certain feature/sensor Please be so kind and make use if the discussion feature of this repo (and do not create an issue) - TIA

### Add a description

ECO mode cannot be activated.
Values are transferred. Configuration is complete.
Any ideas?
I have checked everything several times.
![IMG_6715](https://github.com/marq24/ha-goecharger-api2/assets/35043404/45ce529b-6d9b-43f4-9113-15fe95f05bec)
![IMG_6714](https://github.com/marq24/ha-goecharger-api2/assets/35043404/d7d54715-5bab-4c8b-91ae-3e3fb49fa666)

### Add your DEBUG log output

```shell
2024-06-27 16:42:50.477 INFO (MainThread) [custom_components.goecharger_api2.pygoecharger_ha] going to write {'ids': '{"pGrid": -7699.0, "pPv": 8593.0, "pAkku": 0.0}'} to go-eCharger@192.168.25.124
2024-06-27 16:42:50.521 DEBUG (MainThread) [custom_components.goecharger_api2] write result: {'ids': {'pGrid': -7699.0, 'pPv': 8593.0, 'pAkku': 0.0}}
2024-06-27 16:42:51.335 DEBUG (MainThread) [custom_components.goecharger_api2.pygoecharger_ha] going to request 20 keys from go-eCharger@192.168.25.124
2024-06-27 16:42:51.378 DEBUG (MainThread) [custom_components.goecharger_api2.pygoecharger_ha] read 20 values from go-eCharger@192.168.25.124
2024-06-27 16:42:51.378 DEBUG (MainThread) [custom_components.goecharger_api2] Finished fetching goecharger_api2 data in 0.043 seconds (success: True)
```

**Comment by marq24** [comment]
mhhh - so what happen, when you open (in the go-eCharder App) the Settings for the PV-Surplus charging?! I have to admit, that I can't remember if there was an additional setting option?

In the Charger-Tab I select ECO-Mode and then on the ECO-Tab I just configured the "Enable PV" ON and then just double checked, if the settings in the submenu seams to be fine...

Then (once the HA-Automation started to send data to the wallbox) the "not available" went away after a short while (less then 5 minutes)... 

And the log output you shared - looks like that the automation does what it supposed to do - and the "write result" output also indicates, that the WB have no issues processing the sent data...

One thing you can verify if the WB will provide a plausible `sensor.goe_[SERIAL-HERE]_lpsc_delta` and `sensor.goe_[SERIAL-HERE]_pvopt_averagepgrid` values ?!

**Comment by CjmHo** [comment]
Hm, if my memory doesn't betray me I had the same issue. 
I think I configured the settings under "Flexibler Energietarif" and then I was able to switch on Eco mode. I did NOT switch on the "Flexibler Energietarif" option! 

I find it a tad strange that one needs  to specify "Preisgrenze", "Country", "Anbieter", "Tarif" and so on,  even though  the  "Flexibler Energietarif" switch says "Aus" (off). Probably a bug in the go-echarger software?
